R/30_svds.R

```diff
@@ -165,13 +169,25 @@ svds.dgCMatrix <- function(A, k, nu = k, nv = k, opts = list(), ...)
 ##' @rdname svds
 ##' @export
 svds.dgRMatrix <- function(A, k, nu = k, nv = k, opts = list(), ...)
-    svds_real_gen(A, k, nu, nv, opts, mattype = "dgRMatrix")
+{
+    fun = if(isSymmetric(A)) svds_real_sym else svds_real_gen
+    fun(A, k, nu, nv, opts, mattype = "dgRMatrix")
```

Comment:
i just read above that “isSymmetric doesn’t support dgRMatrix”, but maybe we can convert dgRMatrix to dsRMatrix?